apps/speed.c

```diff
@@ -798,24 +799,26 @@ static int EVP_Update_loop_aead(void *args)
     int outl, count;
     unsigned char aad[13] = { 0xcc };
     unsigned char faketag[16] = { 0xcc };
+    int dummy = 1;
 
     if (decrypt) {
         for (count = 0; COND(c[D_EVP][testnum]); count++) {
-            (void)EVP_DecryptInit_ex(ctx, NULL, NULL, NULL, iv);
-            (void)EVP_CIPHER_CTX_ctrl(ctx, EVP_CTRL_AEAD_SET_TAG,
-                                      sizeof(faketag), faketag);
-            (void)EVP_DecryptUpdate(ctx, NULL, &outl, aad, sizeof(aad));
-            (void)EVP_DecryptUpdate(ctx, buf, &outl, buf, lengths[testnum]);
-            (void)EVP_DecryptFinal_ex(ctx, buf + outl, &outl);
+            dummy *= EVP_DecryptInit_ex(ctx, NULL, NULL, NULL, iv);
+            dummy *= EVP_CIPHER_CTX_ctrl(ctx, EVP_CTRL_AEAD_SET_TAG,
+                                        sizeof(faketag), faketag);
+            dummy *= EVP_DecryptUpdate(ctx, NULL, &outl, aad, sizeof(aad));
+            dummy *= EVP_DecryptUpdate(ctx, buf, &outl, buf, lengths[testnum]);
+            dummy *= EVP_DecryptFinal_ex(ctx, buf + outl, &outl);
         }
     } else {
         for (count = 0; COND(c[D_EVP][testnum]); count++) {
-            (void)EVP_EncryptInit_ex(ctx, NULL, NULL, NULL, iv);
-            (void)EVP_EncryptUpdate(ctx, NULL, &outl, aad, sizeof(aad));
-            (void)EVP_EncryptUpdate(ctx, buf, &outl, buf, lengths[testnum]);
-            (void)EVP_EncryptFinal_ex(ctx, buf + outl, &outl);
+            dummy *= EVP_EncryptInit_ex(ctx, NULL, NULL, NULL, iv);
+            dummy *= EVP_EncryptUpdate(ctx, NULL, &outl, aad, sizeof(aad));
+            dummy *= EVP_EncryptUpdate(ctx, buf, &outl, buf, lengths[testnum]);
+            dummy *= EVP_EncryptFinal_ex(ctx, buf + outl, &outl);
```

Comment:
Ah, so I see the (void) cast is probably insufficient. Then I'd say we should rather not merge this change.

Reply:
Resolving outdated conversation